src/OnlineRetailPortal.Services/Services/ProductService.cs

```diff
@@ -1,22 +1,25 @@
 ﻿using OnlineRetailPortal.Contracts;
-using System;
+using OnlineRetailPortal.Core;
 using System.Collections.Generic;
-using System.Text;
 using System.Threading.Tasks;
 
 namespace OnlineRetailPortal.Services
 {
-    class ProductService : IProductService
+   public class ProductService : IProductService
     {
-       
-        public Task<GetProductsServiceResponse> GetProductsAsync(GetProductsServiceRequest request)
+        public async Task<GetProductsServiceResponse> GetProductsAsync(GetProductsServiceRequest request)
         {
-            throw new NotImplementedException();
+            IProductStore productStore=null;
+            var response = await Core.Product.GetProductsAsync(request,productStore );
+            //get list based on paging info 
+            return await Task.FromResult(response.ToGetProductsContract());
         }
 
-        public Task<GetProductServiceResponse> GetProductAsync(string productId)
+        public async Task<GetProductServiceResponse>  GetProductAsync(string productId)
         {
-            throw new NotImplementedException();
+            IProductStore productStore = null;
+           var response = await Core.Product.GetProductAsync(productId, productStore);
```

Comment:
Why are we sending null productstore?

Reply:
Shubham you need to add a comment that factory or logic to resolve the product store will be handled later ( or whatever applies here)

Reply:
done